examples/wCCD/src/constants.ts

```diff
@@ -22,3 +24,6 @@ export const VIEW_FUNCTION_RAW_SCHEMA =
     '//8CAQAAAAkAAABjaXMyX3dDQ0QBABQAAgAAAAMAAAB1cmwWAgQAAABoYXNoFQIAAAAEAAAATm9uZQIEAAAAU29tZQEBAAAAEyAAAAACAQAAAAQAAAB2aWV3BRQAAwAAAAUAAABhZG1pbhUCAAAABwAAAEFjY291bnQBAQAAAAsIAAAAQ29udHJhY3QBAQAAAAwGAAAAcGF1c2VkAQwAAABtZXRhZGF0YV91cmwUAAIAAAADAAAAdXJsFgEEAAAAaGFzaBUCAAAABAAAAE5vbmUCBAAAAFNvbWUBAQAAABMgAAAAAhUEAAAADgAAAEludmFsaWRUb2tlbklkAhEAAABJbnN1ZmZpY2llbnRGdW5kcwIMAAAAVW5hdXRob3JpemVkAgYAAABDdXN0b20BAQAAABUJAAAACwAAAFBhcnNlUGFyYW1zAgcAAABMb2dGdWxsAgwAAABMb2dNYWxmb3JtZWQCDgAAAENvbnRyYWN0UGF1c2VkAhMAAABJbnZva2VDb250cmFjdEVycm9yAhMAAABJbnZva2VUcmFuc2ZlckVycm9yAhoAAABGYWlsZWRVcGdyYWRlTWlzc2luZ01vZHVsZQIcAAAARmFpbGVkVXBncmFkZU1pc3NpbmdDb250cmFjdAIlAAAARmFpbGVkVXBncmFkZVVuc3VwcG9ydGVkTW9kdWxlVmVyc2lvbgI=';
 export const BALANCEOF_FUNCTION_RAW_SCHEMA =
     '//8CAQAAAAkAAABjaXMyX3dDQ0QBABQAAgAAAAMAAAB1cmwWAgQAAABoYXNoFQIAAAAEAAAATm9uZQIEAAAAU29tZQEBAAAAEyAAAAACAQAAAAkAAABiYWxhbmNlT2YGEAEUAAIAAAAIAAAAdG9rZW5faWQdAAcAAABhZGRyZXNzFQIAAAAHAAAAQWNjb3VudAEBAAAACwgAAABDb250cmFjdAEBAAAADBABGyUAAAAVBAAAAA4AAABJbnZhbGlkVG9rZW5JZAIRAAAASW5zdWZmaWNpZW50RnVuZHMCDAAAAFVuYXV0aG9yaXplZAIGAAAAQ3VzdG9tAQEAAAAVCQAAAAsAAABQYXJzZVBhcmFtcwIHAAAATG9nRnVsbAIMAAAATG9nTWFsZm9ybWVkAg4AAABDb250cmFjdFBhdXNlZAITAAAASW52b2tlQ29udHJhY3RFcnJvcgITAAAASW52b2tlVHJhbnNmZXJFcnJvcgIaAAAARmFpbGVkVXBncmFkZU1pc3NpbmdNb2R1bGUCHAAAAEZhaWxlZFVwZ3JhZGVNaXNzaW5nQ29udHJhY3QCJQAAAEZhaWxlZFVwZ3JhZGVVbnN1cHBvcnRlZE1vZHVsZVZlcnNpb24C';
+
+export const WALLET_CONNECT_PROJECT_ID = '76324905a70fe5c388bab46d3e0564dc';
+export const TESTNET = new Network('testnet', TESTNET_GENESIS_BLOCK_HASH, 'https://json-rpc.testnet.concordium.com');
```

Comment:
If the user tries to connect but its browser wallet is set to mainnet, the connection button does not work. Could we add an error message, explaining to the user that it needs to switch its browser wallet to testnet to get the connection button to work.

If a user connects their wallet to testnet and then later switches to mainnet and tries to use the wCCD front-end again the following error is shown: "Site is not whitelisted". Could we add some better error message guiding the user back to what he/she has to do to be able to use the front-end again.

Reply:
@DOBEN Are these issues introduced by this PR?

I don't have mainnet accounts in my browser wallet, but as far as I can tell there is no way to tell the network settings of the browser from the app? @shjortConcordium 

There is a handler for `chainChanged` events which I would expect to trigger when the settings change, but it doesn't seem to be the case... That handler would check the genesis hash against the expected one and raise an alert like it was before.

Reply:
There is an issue, where the event doesn't trigger if you don't have an account on the network that you change to, but otherwise it should trigger.

Reply:
And you can at https://github.com/Concordium/concordium-browser-wallet/pull/201#discussion_r1047181296 where it previously checked for the network.

Reply:
That check is still happening in `onChainChanged` in `WithWalletConnection`.

Reply:
I checked again and as @shjortConcordium described the edge case happens if the user is on mainnet (with no mainnet account) then the `Connect Browser Wallet` button does not work (and we have no error message that would hint to the user to switch to testnet). A similar edge case is when the user switches to mainnet  (with no mainnet account) while already connected to the side. 

Feel free to resolve this thread if you think that we would first have to add some events/getter functions in the wallet that would support this feature maybe.

Reply:
I filed https://concordium.atlassian.net/browse/CBW-633 with suggestions to improve the BW to improve this situation. I think the remaining issues are edge case annoyances that we can live with for a short while, so I'll merge this now and look into improving the error message in a followup. This allows us to get this feature out and make managers happy again :)

I'll also address a minor issue that the BW doesn't really support disconnecting, so when we "do" that the event handlers hang on and get reregistered on a "new" connection. This causes them to fire as many times as the connection has been reset.

@shjortConcordium If you continue to see connection issues, please report it as an issue or comment it on the next PR.